app_dart/lib/src/request_handlers/get_timeseries_history.dart

```diff
@@ -0,0 +1,117 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:appengine/appengine.dart';
+import 'package:cocoon_service/src/model/appengine/benchmark_data.dart';
+import 'package:cocoon_service/src/model/appengine/time_series_value.dart';
+import 'package:gcloud/db.dart';
+import 'package:meta/meta.dart';
+
+import '../datastore/cocoon_config.dart';
+import '../model/appengine/commit.dart';
+import '../model/appengine/key_helper.dart';
+import '../model/appengine/time_series.dart';
+import '../model/appengine/time_series_entity.dart';
+import '../request_handling/body.dart';
+import '../request_handling/exceptions.dart';
+import '../request_handling/regular_request_handler.dart';
+import '../service/datastore.dart';
+
+@immutable
+class GetTimeSeriesHistory extends RegularRequestHandler<GetTimeSeriesHistoryResponse> {
+  const GetTimeSeriesHistory(
+    Config config, {
+    @visibleForTesting DatastoreServiceProvider datastoreProvider,
+  })  : datastoreProvider =
+            datastoreProvider ?? DatastoreService.defaultProvider,
+        super(config: config);
+
+  final DatastoreServiceProvider datastoreProvider;
+  static const String timeSeriesKeyParam = 'TimeSeriesKey';
+  static const String startFromParam = 'StartFrom';
+
+  @override
+  Future<GetTimeSeriesHistoryResponse> post() async {
+    checkRequiredParameters(<String>[timeSeriesKeyParam]);
+    const int maxRecords = 1500;
+    final DatastoreService datastore = datastoreProvider();
+    final KeyHelper keyHelper = KeyHelper(applicationContext: AppEngineContext(false,'','','','','',Uri()));
+    final Set<Commit> commits =
+        await datastore.queryRecentCommits(limit: maxRecords).toSet();
+
+    Key timeSeriesKey;
+    try {
+      timeSeriesKey = keyHelper.decode(requestData[timeSeriesKeyParam]);
+    } catch (error) {
+      throw BadRequestException(
+          'Bad timeSeries key: ${requestData[timeSeriesKeyParam]}');
+    }
```

Comment:
Would the exception here make the backend crash?

Reply:
It shouldn't.

Reply:
This should just make the backend return a 4xx error to the client